### PR TITLE
Let Http2ConnectionHandler close stream with voidPromise

### LINKED ISSUE #10816: Cannot write/flush HTTP/2 with voidPromise
### Expected behavior
Can `write` / `writeAndFlush` frames that may close the stream from a HTTP/2 context

### Actual behavior
`Http2ConnectionHandler#closeStream` tries to `addListener` to the future even if it's void. If void, that will fail.

### Steps to reproduce
`writeAndFlush` a Http2DataFrame to the context with Http2FrameCodec present

### Minimal yet complete reproducer code (or URL to code)
Ran into this using netty [from Clojure](https://github.com/valerauko/iny/blob/7e71ddf649d5b64626139beb784eabf9b5ad88a5/src/iny/http2/handler.clj#L86-L98).

### Netty version
4.1.54

### PR DESCRIPTION
**Pull request by valerauko**
Motivation:

Http2ConnectionHandler tries to addListener to the future without checking if it's void. If it is void, this will fail and generate an exception. 
 
Modifications:

Unvoid the promise before attempting to write
 
Result:

Fixes #10816
Writing with a voidPromise no longer generates exceptions.

### CODE REVIEW

File: codec-http2/src/test/java/io/netty/handler/codec/http2/Http2ConnectionHandlerTest.java [hunk 1]
```diff
@@ -691,6 +691,14 @@ public ChannelFuture answer(InvocationOnMock invocation) throws Throwable {
         verify(pipeline).fireExceptionCaught(cause);
     }
 
+    @Test
+    public void canCloseStreamWithVoidPromise() throws Exception {
+        handler = newHandler();
+        handler.closeStream(stream, ctx.voidPromise());
+        verify(stream, times(1)).close();
+        verifyNoMoreInteractions(stream);
```
Comment (valerauko) [inline codec-http2/src/test/java/io/netty/handler/codec/http2/Http2ConnectionHandlerTest.java:699]: Based on the other tests I wanted to do something like this:
```java
        verify(voidPromise, never()).addListener(any(GenericFutureListener.class));
        verify(ctx.channel(), times(1)).newPromise();
        verify(promise, times(1)).addListener(any(GenericFutureListener.class));
```
I'm not used to this testing suite, so please advise how can I get these checks to work... :pray:

**Comment by normanmaurer** [comment]
I think this fix is not correct as the new promise you create will never be notified. Can you show me the full stack trace when the error shows up ?

**Comment by valerauko** [comment]
Posted the stacktrace to the issue. I realized that DefaultHttp2ConnectionEncoder#writeData doesn't have the unvoid that writeHeaders0 has. Adding the unvoid there solved the errors _and_ notified the promise as it should.

**Comment by normanmaurer** [comment]
@netty-bot test this please

**Comment by normanmaurer** [comment]
@valerauko thanks a lot

**Pull request merged**